### ISSUE TITLE
-m pip install does not work

### ISSUE THREAD

**Issue by tekknolagi** [tekknolagi/skybison#14]
`./build-relwithdebinfo/bin/python -m ensurepip` works fine.

`./build-relwithdebinfo/bin/python -m pip install pycca`, for example, crashes with:

```
Stack (most recent call first):
  File "", line ??? in _unimplemented
  File "library/_io.py", line 972 in _readinto
  File "library/_io.py", line 634 in readinto
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/skybison3.8/http/client.py", line 502 in readinto
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/skybison3.8/http/client.py", line 458 in read
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/_vendor/cachecontrol/filewrapper.py", line 62 in read
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/_vendor/urllib3/response.py", line 507 in read
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/_vendor/urllib3/response.py", line 564 in stream
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/_vendor/requests/models.py", line 751 in generate
  File "library/builtins.py", line 2581 in <listcomp>
  File "library/builtins.py", line 2581 in join
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/_vendor/requests/models.py", line 829 in content
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/_vendor/requests/sessions.py", line 683 in send
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/_vendor/requests/sessions.py", line 530 in request
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/_internal/network/session.py", line 421 in request
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/_vendor/requests/sessions.py", line 543 in get
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/_internal/index/collector.py", line 149 in _get_html_response
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/_internal/index/collector.py", line 451 in _get_html_page
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/_internal/index/collector.py", line 612 in fetch_page
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/_internal/index/package_finder.py", line 792 in process_project_url
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/_internal/index/package_finder.py", line 827 in find_all_candidates
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/_internal/index/package_finder.py", line 883 in find_best_candidate
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/_internal/index/package_finder.py", line 900 in find_requirement
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/_internal/resolution/legacy/resolver.py", line 279 in _populate_link
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/_internal/resolution/legacy/resolver.py", line 313 in _get_abstract_dist_for
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/_internal/resolution/legacy/resolver.py", line 362 in _resolve_one
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/_internal/resolution/legacy/resolver.py", line 179 in resolve
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/_internal/commands/install.py", line 332 in run
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/_internal/cli/req_command.py", line 185 in wrapper
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/_internal/cli/base_command.py", line 188 in _main
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/_internal/cli/base_command.py", line 114 in main
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/_internal/cli/main.py", line 75 in main
  File "/home/max/Documents/code/skybison/build-relwithdebinfo/lib/python3.8/site-packages/pip/__main__.py", line 26 in <module>
  File "", line ??? in _exec
  File "library/builtins.py", line 3717 in exec
  File "third-party/cpython/Lib/runpy.py", line 87 in _run_code
  File "third-party/cpython/Lib/runpy.py", line 194 in _run_module_as_main

'_unimplemented' called in function '_readinto'

use env PYRO_RAISE_ON_UNIMPLEMENTED=1 or
sys.PYRO_RAISE_ON_UNIMPLEMENTED=True to raise instead of abort.
```

Where `BufferedReader._readinto` is not implemented. In this particular call, the `buf` parameter is a `memoryview`. With this patch (which is not fast, and may not be correct):

```diff
diff --git a/library/_io.py b/library/_io.py
index 2a9e3d7d..4dffd904 100644
--- a/library/_io.py
+++ b/library/_io.py
@@ -969,7 +969,7 @@ class BufferedReader(_BufferedIOMixin, bootstrap=True):
         return _BufferedIOMixin.tell(self) - self._buffer_num_bytes + self._read_pos
 
     def _readinto(self, buf, read1):
-        _unimplemented()
+        buf[:] = self.read1()
 
     def __init__(self, raw, buffer_size=DEFAULT_BUFFER_SIZE):
         if not raw.readable():
diff --git a/library/builtins.py b/library/builtins.py
index 7ddcbc20..a869735a 100644
--- a/library/builtins.py
+++ b/library/builtins.py
@@ -5469,7 +5469,8 @@ class memoryview(bootstrap=True):
         pass
 
     def tobytes(self):
-        _unimplemented()
+        return bytes(self)
+        # _unimplemented()
 
     def tolist(self):
         _memoryview_guard(self)
```

we get further:

```
cedar% ./build-relwithdebinfo/bin/python -m pip install pycca
Collecting pycca
  Downloading pycca-0.2.0.tar.gz (32 kB)
ERROR: THESE PACKAGES DO NOT MATCH THE HASHES FROM THE REQUIREMENTS FILE. If you have updated the package versions, please update the hashes. Otherwise, examine the package contents carefully; someone may have tampered with them.
    pycca from https://files.pythonhosted.org/packages/85/22/d291891657125ce473e1bd89d1c571162c5508278c0f814ff3a7e09b67f1/pycca-0.2.0.tar.gz#sha256=2f9d79b8a7b997bd0b0bc38ab84fc575eb0725111441ec5bf27e013a12b46abb:
        Expected sha256 2f9d79b8a7b997bd0b0bc38ab84fc575eb0725111441ec5bf27e013a12b46abb
             Got        5c6375944ee2f349aa5d6b1218980781ed1a14c89b2740b801420389fa4a4eb6
```

but still no dice.

**Comment by tekknolagi** [comment]
Turns out there is a lot going on here  🙃

**Comment by tekknolagi** [comment]
Fixed in #18 and ff1aefbd45db32dc8d0dfe5409884b5944621501
